lib/output.js

```diff
@@ -118,6 +118,48 @@ function toBuffer (options, callback) {
   return this._pipeline(is.fn(options) ? options : callback);
 }
 
+/**
+ * Same as toBuffer, but writes output to a Uint8Array.
+ *
+ * @example
+ * sharp(input)
+ *   .toUint8Array((err, data, info) => { ... });
+ *
+ * @example
+ * sharp(input)
+ *   .toUint8Array()
+ *   .then(data => { ... })
+ *   .catch(err => { ... });
+ *
+ * @example
+ * sharp(input)
+ *   .toUint8Array({ resolveWithObject: true })
+ *   .then(({ data, info }) => { ... })
+ *   .catch(err => { ... });
+ *
+ * @param {Object|Function} [optionsOrCallback] The resolveWithObject option if you want to the promise to include info.
+ * @param {boolean} [options.resolveWithObject] Resolve the Promise with an Object containing `data` and `info` properties instead of resolving only with `data`.
+ * @returns {Promise<Uint8Array>} - when no callback is provided
+ */
+function toUint8Array (optionsOrCallback) {
+  // callback mode
+  if (is.fn(optionsOrCallback)) {
+    // wrap callback with function that converts to Uint8Array
+    return this.toBuffer(optionsOrCallback, (err, data, info) => optionsOrCallback(err, new Uint8Array(data.buffer), info));
```

Comment:
Are there too many callbacks here?
```diff
- return this.toBuffer(optionsOrCallback, (err, data, info) => optionsOrCallback(err, new Uint8Array(data.buffer), info));
+ return this.toBuffer((err, data, info) => optionsOrCallback(err, new Uint8Array(data.buffer), info));
```

Reply:
Don't know how I missed that. good catch.